app/src/main/java/com/github/multimatum_team/multimatum/DeadlineNotification.kt

```diff
@@ -34,22 +35,49 @@ class DeadlineNotification {
     }
 
     /**
-    Set a notification that will be triggered in a given time in ms.
-    you can pass a title/description and Id in parameter
-    */
-    fun setNotification(deadline: Deadline, context: Context, triggerTime: Long){
+     * Set a notification for a deadline in a given time before it is due.
+     *
+     * For example, to set an alarm (=notif) 1 hours before the deadline is due, use "setNotification(Pair(deadline_id, deadline), context, Duration.of(1, ChronoUnit.HOURS).toMillis())
+     * @param timeBeforeDeadline : time in ms before the deadline, when the alarm has to be triggered
+     *
+     * @return void
+     */
+    fun setNotification(deadlinePair: Pair<DeadlineID, Deadline>, context: Context, timeBeforeDeadline: Long){
+        val id: DeadlineID = deadlinePair.first
```

Comment:
You can use destructuring syntax e.g. `val (id, deadline) = deadlinePair`.
But in this case I think it would make more sense to pass two distinct parameters like `fun setNotification(id: DeadlineID, deadline: Deadline, ...)`.

Reply:
It doesn't change that much. Since we retrieve pair from firebase, I prefer doing it like that